### PR TITLE
HADOOP-18180. Replaces ExecutorServicePool with ExecutorService

### PR DESCRIPTION
**Pull request by ahmarsuhail**
### Description of PR

The `ExecutorServicePool` class was only submitting tasks to the `ExecutorService`. This PR removes `ExecutorServicePool` and submits directly to the `ExecutorService`.

### How was this patch tested?

Tested in eu-west-1 by running

`mvn -Dparallel-tests -DtestsThreadCount=16 clean verify`

### CODE REVIEW

File: hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java [hunk 1]
```diff
@@ -1573,7 +1560,7 @@ protected S3AReadOpContext createReadContext(
         statistics,
         statisticsContext,
         fileStatus,
-        futurePool,
+        unboundedThreadPool,
```
Comment (dannycjones) [inline hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java]: Why not the `boundedThreadPool` originally wrapped by `futurePool`?

File: hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AReadOpContext.java [hunk 2]
```diff
@@ -18,10 +18,11 @@
 
 package org.apache.hadoop.fs.s3a;
 
+import java.util.concurrent.ExecutorService;
```
Comment (dannycjones) [inline hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AReadOpContext.java:21]: There's a `javax.annotation.Nullable` import below Apache. Can you bring it up to this import?

rel: https://github.com/steveloughran/formality/blob/master/styleguide/styleguide.md#imports

**Review by dannycjones — CHANGES_REQUESTED** [review]
I have a concern on which thread pool should be passed by S3AFileSystem; otherwise, looking good.

File: hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java (hunk 1 above)
Comment (ahmarsuhail) [reply hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java]: that's my bad, it should be `boundedThreadPool`. Have updated

**Review by dannycjones — APPROVED** [review]
+1 (non-binding) assuming Yetus is happy

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |  13m  8s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 5 new or modified test files.  |
|||| _ feature-HADOOP-18028-s3a-prefetch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  38m 42s |  |  feature-HADOOP-18028-s3a-prefetch passed  |
| +1 :green_heart: |  compile  |   1m  2s |  |  feature-HADOOP-18028-s3a-prefetch passed with JDK Private Build-11.0.15+10-Ubuntu-0ubuntu0.20.04.1  |
| +1 :green_heart: |  compile  |   0m 56s |  |  feature-HADOOP-18028-s3a-prefetch passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  checkstyle  |   0m 49s |  |  feature-HADOOP-18028-s3a-prefetch passed  |
| +1 :green_heart: |  mvnsite  |   1m  1s |  |  feature-HADOOP-18028-s3a-prefetch passed  |
| +1 :green_heart: |  javadoc  |   0m 50s |  |  feature-HADOOP-18028-s3a-prefetch passed with JDK Private Build-11.0.15+10-Ubuntu-0ubuntu0.20.04.1  |
| +1 :green_heart: |  javadoc  |   0m 52s |  |  feature-HADOOP-18028-s3a-prefetch passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  spotbugs  |   1m 38s |  |  feature-HADOOP-18028-s3a-prefetch passed  |
| +1 :green_heart: |  shadedclient  |  22m  0s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 44s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 48s |  |  the patch passed with JDK Private Build-11.0.15+10-Ubuntu-0ubuntu0.20.04.1  |
| +1 :green_heart: |  javac  |   0m 48s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 39s |  |  the patch passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  javac  |   0m 39s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   0m 26s |  |  hadoop-tools/hadoop-aws: The patch generated 0 new + 3 unchanged - 2 fixed = 3 total (was 5)  |
| +1 :green_heart: |  mvnsite  |   0m 43s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 23s |  |  the patch passed with JDK Private Build-11.0.15+10-Ubuntu-0ubuntu0.20.04.1  |
| +1 :green_heart: |  javadoc  |   0m 35s |  |  the patch passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  spotbugs  |   1m 12s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  21m 17s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  |   2m 40s | [/patch-unit-hadoop-tools_hadoop-aws.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4520/1/artifact/out/patch-unit-hadoop-tools_hadoop-aws.txt) |  hadoop-aws in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 49s |  |  The patch does not generate ASF License warnings.  |
|  |   | 112m 28s |  |  |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.fs.s3a.read.TestS3InputStream |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4520/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/4520 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell |
| uname | Linux caff41b93769 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | feature-HADOOP-18028-s3a-prefetch / 225f0f014e53e30d7f3592632a22136cd312e302 |
| Default Java | Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Private Build-11.0.15+10-Ubuntu-0ubuntu0.20.04.1 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4520/1/testReport/ |
| Max. process+thread count | 716 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-aws U: hadoop-tools/hadoop-aws |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4520/1/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

**Comment by ahmarsuhail** [comment]
Failing unit tests are fixed in https://github.com/apache/hadoop/pull/4386

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |  17m 39s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 5 new or modified test files.  |
|||| _ feature-HADOOP-18028-s3a-prefetch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  40m 41s |  |  feature-HADOOP-18028-s3a-prefetch passed  |
| +1 :green_heart: |  compile  |   0m 55s |  |  feature-HADOOP-18028-s3a-prefetch passed with JDK Private Build-11.0.15+10-Ubuntu-0ubuntu0.20.04.1  |
| +1 :green_heart: |  compile  |   0m 47s |  |  feature-HADOOP-18028-s3a-prefetch passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  checkstyle  |   0m 42s |  |  feature-HADOOP-18028-s3a-prefetch passed  |
| +1 :green_heart: |  mvnsite  |   0m 55s |  |  feature-HADOOP-18028-s3a-prefetch passed  |
| +1 :green_heart: |  javadoc  |   0m 40s |  |  feature-HADOOP-18028-s3a-prefetch passed with JDK Private Build-11.0.15+10-Ubuntu-0ubuntu0.20.04.1  |
| +1 :green_heart: |  javadoc  |   0m 42s |  |  feature-HADOOP-18028-s3a-prefetch passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  spotbugs  |   1m 29s |  |  feature-HADOOP-18028-s3a-prefetch passed  |
| +1 :green_heart: |  shadedclient  |  23m 40s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 38s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 44s |  |  the patch passed with JDK Private Build-11.0.15+10-Ubuntu-0ubuntu0.20.04.1  |
| +1 :green_heart: |  javac  |   0m 44s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 34s |  |  the patch passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  javac  |   0m 34s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   0m 23s |  |  hadoop-tools/hadoop-aws: The patch generated 0 new + 3 unchanged - 2 fixed = 3 total (was 5)  |
| +1 :green_heart: |  mvnsite  |   0m 40s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 21s |  |  the patch passed with JDK Private Build-11.0.15+10-Ubuntu-0ubuntu0.20.04.1  |
| +1 :green_heart: |  javadoc  |   0m 30s |  |  the patch passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  spotbugs  |   1m 15s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  23m 31s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  |   2m 41s | [/patch-unit-hadoop-tools_hadoop-aws.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4520/2/artifact/out/patch-unit-hadoop-tools_hadoop-aws.txt) |  hadoop-aws in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 42s |  |  The patch does not generate ASF License warnings.  |
|  |   | 120m 53s |  |  |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.fs.s3a.read.TestS3InputStream |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4520/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/4520 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell |
| uname | Linux 8eb241625259 4.15.0-175-generic #184-Ubuntu SMP Thu Mar 24 17:48:36 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | feature-HADOOP-18028-s3a-prefetch / 3bd7b73823eb9713c259932e4133ce6b5190e11e |
| Default Java | Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Private Build-11.0.15+10-Ubuntu-0ubuntu0.20.04.1 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4520/2/testReport/ |
| Max. process+thread count | 566 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-aws U: hadoop-tools/hadoop-aws |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4520/2/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |  15m 17s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 5 new or modified test files.  |
|||| _ feature-HADOOP-18028-s3a-prefetch Compile Tests _ |
| -1 :x: |  mvninstall  |   0m 32s | [/branch-mvninstall-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4520/3/artifact/out/branch-mvninstall-root.txt) |  root in feature-HADOOP-18028-s3a-prefetch failed.  |
| -1 :x: |  compile  |   0m 31s | [/branch-compile-hadoop-tools_hadoop-aws-jdkPrivateBuild-11.0.15+10-Ubuntu-0ubuntu0.20.04.1.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4520/3/artifact/out/branch-compile-hadoop-tools_hadoop-aws-jdkPrivateBuild-11.0.15+10-Ubuntu-0ubuntu0.20.04.1.txt) |  hadoop-aws in feature-HADOOP-18028-s3a-prefetch failed with JDK Private Build-11.0.15+10-Ubuntu-0ubuntu0.20.04.1.  |
| -1 :x: |  compile  |   0m 32s | [/branch-compile-hadoop-tools_hadoop-aws-jdkPrivateBuild-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4520/3/artifact/out/branch-compile-hadoop-tools_hadoop-aws-jdkPrivateBuild-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07.txt) |  hadoop-aws in feature-HADOOP-18028-s3a-prefetch failed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07.  |
| -0 :warning: |  checkstyle  |   0m 30s | [/buildtool-branch-checkstyle-hadoop-tools_hadoop-aws.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4520/3/artifact/out/buildtool-branch-checkstyle-hadoop-tools_hadoop-aws.txt) |  The patch fails to run checkstyle in hadoop-aws  |
| -1 :x: |  mvnsite  |   0m 32s | [/branch-mvnsite-hadoop-tools_hadoop-aws.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4520/3/artifact/out/branch-mvnsite-hadoop-tools_hadoop-aws.txt) |  hadoop-aws in feature-HADOOP-18028-s3a-prefetch failed.  |
| -1 :x: |  javadoc  |   0m 31s | [/branch-javadoc-hadoop-tools_hadoop-aws-jdkPrivateBuild-11.0.15+10-Ubuntu-0ubuntu0.20.04.1.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4520/3/artifact/out/branch-javadoc-hadoop-tools_hadoop-aws-jdkPrivateBuild-11.0.15+10-Ubuntu-0ubuntu0.20.04.1.txt) |  hadoop-aws in feature-HADOOP-18028-s3a-prefetch failed with JDK Private Build-11.0.15+10-Ubuntu-0ubuntu0.20.04.1.  |
| -1 :x: |  javadoc  |   0m 31s | [/branch-javadoc-hadoop-tools_hadoop-aws-jdkPrivateBuild-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4520/3/artifact/out/branch-javadoc-hadoop-tools_hadoop-aws-jdkPrivateBuild-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07.txt) |  hadoop-aws in feature-HADOOP-18028-s3a-prefetch failed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07.  |
| -1 :x: |  spotbugs  |   0m 31s | [/branch-spotbugs-hadoop-tools_hadoop-aws.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4520/3/artifact/out/branch-spotbugs-hadoop-tools_hadoop-aws.txt) |  hadoop-aws in feature-HADOOP-18028-s3a-prefetch failed.  |
| +1 :green_heart: |  shadedclient  |   3m 37s |  |  branch has no errors when building and testing our client artifacts.  |
| -0 :warning: |  patch  |   4m  9s |  |  Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary.  |
|||| _ Patch Compile Tests _ |
| -1 :x: |  mvninstall  |   0m 55s | [/patch-mvninstall-hadoop-tools_hadoop-aws.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4520/3/artifact/out/patch-mvninstall-hadoop-tools_hadoop-aws.txt) |  hadoop-aws in the patch failed.  |
| -1 :x: |  compile  |   4m  0s | [/patch-compile-hadoop-tools_hadoop-aws-jdkPrivateBuild-11.0.15+10-Ubuntu-0ubuntu0.20.04.1.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4520/3/artifact/out/patch-compile-hadoop-tools_hadoop-aws-jdkPrivateBuild-11.0.15+10-Ubuntu-0ubuntu0.20.04.1.txt) |  hadoop-aws in the patch failed with JDK Private Build-11.0.15+10-Ubuntu-0ubuntu0.20.04.1.  |
| -1 :x: |  javac  |   4m  0s | [/patch-compile-hadoop-tools_hadoop-aws-jdkPrivateBuild-11.0.15+10-Ubuntu-0ubuntu0.20.04.1.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4520/3/artifact/out/patch-compile-hadoop-tools_hadoop-aws-jdkPrivateBuild-11.0.15+10-Ubuntu-0ubuntu0.20.04.1.txt) |  hadoop-aws in the patch failed with JDK Private Build-11.0.15+10-Ubuntu-0ubuntu0.20.04.1.  |
| -1 :x: |  compile  |   1m 36s | [/patch-compile-hadoop-tools_hadoop-aws-jdkPrivateBuild-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4520/3/artifact/out/patch-compile-hadoop-tools_hadoop-aws-jdkPrivateBuild-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07.txt) |  hadoop-aws in the patch failed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07.  |
| -1 :x: |  javac  |   1m 36s | [/patch-compile-hadoop-tools_hadoop-aws-jdkPrivateBuild-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4520/3/artifact/out/patch-compile-hadoop-tools_hadoop-aws-jdkPrivateBuild-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07.txt) |  hadoop-aws in the patch failed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07.  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -0 :warning: |  checkstyle  |   0m 29s | [/buildtool-patch-checkstyle-hadoop-tools_hadoop-aws.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4520/3/artifact/out/buildtool-patch-checkstyle-hadoop-tools_hadoop-aws.txt) |  The patch fails to run checkstyle in hadoop-aws  |
| -1 :x: |  mvnsite  |   1m  7s | [/patch-mvnsite-hadoop-tools_hadoop-aws.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4520/3/artifact/out/patch-mvnsite-hadoop-tools_hadoop-aws.txt) |  hadoop-aws in the patch failed.  |
| +1 :green_heart: |  javadoc  |   0m 42s |  |  the patch passed with JDK Private Build-11.0.15+10-Ubuntu-0ubuntu0.20.04.1  |
| -1 :x: |  javadoc  |   0m 38s | [/results-javadoc-javadoc-hadoop-tools_hadoop-aws-jdkPrivateBuild-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4520/3/artifact/out/results-javadoc-javadoc-hadoop-tools_hadoop-aws-jdkPrivateBuild-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07.txt) |  hadoop-tools_hadoop-aws-jdkPrivateBuild-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07 with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07 generated 40 new + 0 unchanged - 0 fixed = 40 total (was 0)  |
| -1 :x: |  spotbugs  |   0m 31s | [/patch-spotbugs-hadoop-tools_hadoop-aws.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4520/3/artifact/out/patch-spotbugs-hadoop-tools_hadoop-aws.txt) |  hadoop-aws in the patch failed.  |
| -1 :x: |  shadedclient  |   6m 52s |  |  patch has errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  |   0m 30s | [/patch-unit-hadoop-tools_hadoop-aws.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4520/3/artifact/out/patch-unit-hadoop-tools_hadoop-aws.txt) |  hadoop-aws in the patch failed.  |
| +0 :ok: |  asflicense  |   0m 31s |  |  ASF License check generated no output?  |
|  |   |  36m 52s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4520/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/4520 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell |
| uname | Linux 7d5b7ab1478f 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | feature-HADOOP-18028-s3a-prefetch / d3f6d6bba110ee4187fc1cd8e3ded1a0a3e20a23 |
| Default Java | Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Private Build-11.0.15+10-Ubuntu-0ubuntu0.20.04.1 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4520/3/testReport/ |
| Max. process+thread count | 88 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-aws U: hadoop-tools/hadoop-aws |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4520/3/console |
| versions | git=2.25.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.14.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 43s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 5 new or modified test files.  |
|||| _ feature-HADOOP-18028-s3a-prefetch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  41m 17s |  |  feature-HADOOP-18028-s3a-prefetch passed  |
| +1 :green_heart: |  compile  |   1m 10s |  |  feature-HADOOP-18028-s3a-prefetch passed with JDK Private Build-11.0.15+10-Ubuntu-0ubuntu0.20.04.1  |
| +1 :green_heart: |  compile  |   1m  1s |  |  feature-HADOOP-18028-s3a-prefetch passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  checkstyle  |   0m 56s |  |  feature-HADOOP-18028-s3a-prefetch passed  |
| +1 :green_heart: |  mvnsite  |   1m 14s |  |  feature-HADOOP-18028-s3a-prefetch passed  |
| +1 :green_heart: |  javadoc  |   0m 50s |  |  feature-HADOOP-18028-s3a-prefetch passed with JDK Private Build-11.0.15+10-Ubuntu-0ubuntu0.20.04.1  |
| +1 :green_heart: |  javadoc  |   0m 56s |  |  feature-HADOOP-18028-s3a-prefetch passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  spotbugs  |   1m 56s |  |  feature-HADOOP-18028-s3a-prefetch passed  |
| -1 :x: |  shadedclient  |  29m 48s |  |  branch has errors when building and testing our client artifacts.  |
| -0 :warning: |  patch  |  30m 27s |  |  Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary.  |
|||| _ Patch Compile Tests _ |
| -1 :x: |  mvninstall  |   0m 31s | [/patch-mvninstall-hadoop-tools_hadoop-aws.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4520/4/artifact/out/patch-mvninstall-hadoop-tools_hadoop-aws.txt) |  hadoop-aws in the patch failed.  |
| +1 :green_heart: |  compile  |   0m 53s |  |  the patch passed with JDK Private Build-11.0.15+10-Ubuntu-0ubuntu0.20.04.1  |
| +1 :green_heart: |  javac  |   0m 53s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 52s |  |  the patch passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  javac  |   0m 52s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -0 :warning: |  checkstyle  |   0m 30s | [/buildtool-patch-checkstyle-hadoop-tools_hadoop-aws.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4520/4/artifact/out/buildtool-patch-checkstyle-hadoop-tools_hadoop-aws.txt) |  The patch fails to run checkstyle in hadoop-aws  |
| -1 :x: |  mvnsite  |   0m 29s | [/patch-mvnsite-hadoop-tools_hadoop-aws.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4520/4/artifact/out/patch-mvnsite-hadoop-tools_hadoop-aws.txt) |  hadoop-aws in the patch failed.  |
| +1 :green_heart: |  javadoc  |   0m 26s |  |  the patch passed with JDK Private Build-11.0.15+10-Ubuntu-0ubuntu0.20.04.1  |
| +1 :green_heart: |  javadoc  |   0m 35s |  |  the patch passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  spotbugs  |   1m 36s |  |  the patch passed  |
| -1 :x: |  shadedclient  |   9m 44s |  |  patch has errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  |   0m 41s | [/patch-unit-hadoop-tools_hadoop-aws.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4520/4/artifact/out/patch-unit-hadoop-tools_hadoop-aws.txt) |  hadoop-aws in the patch failed.  |
| +0 :ok: |  asflicense  |   0m 38s |  |  ASF License check generated no output?  |
|  |   |  97m 56s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4520/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/4520 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell |
| uname | Linux 45ab6f2bb7f9 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | feature-HADOOP-18028-s3a-prefetch / 9ddad61a9f051b9cc3e7aeb538f75a9f98736cf7 |
| Default Java | Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Private Build-11.0.15+10-Ubuntu-0ubuntu0.20.04.1 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4520/4/testReport/ |
| Max. process+thread count | 411 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-aws U: hadoop-tools/hadoop-aws |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4520/4/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

**Pull request closed without merging**